component/admin/updates/3.0.3.php

```diff
@@ -18,6 +18,44 @@
  */
 class RedshopUpdate303 extends RedshopInstallUpdate
 {
+	/**
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function installShopperGroup()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true)
+			->select('*')
+			->from($db->qn('#__redshop_tax_rate'));
+		$taxRates = $db->setQuery($query)->loadAssocList();
```

Comment:
@hoang-tran-redweb incase you need update code for reduce loops, please also using `Redshop\DB\Tool::safeSelect` instead of `$db->loadAssocList()`
cc: @ducredweb

Reply:
@anhredweb I fixed